src/System.IO.Packaging/src/System/IO/Packaging/Package.cs

```diff
@@ -0,0 +1,1255 @@
+                    // These will get flushed to the disk as a part of the DoFlush operation
+                    if (_packageProperties != null)
+                        _packageProperties.Close();
+
+                    // flush relationships
+                    FlushRelationships();
+
+                    //Write out the Relationship XML for the parts
+                    //These streams will get flushed in the DoClose operation.
+                    DoOperationOnEachPart(DoCloseRelationshipsXml);
+
+                    // Close all the parts that are currently open
+                    DoOperationOnEachPart(DoClose);
+
+                    Dispose(true);
+                }
+                finally
+                {
+                    // do this no matter what (handles case of poorly behaving subclass that doesn't call back into Dispose(bool)
+                    _disposed = true;
+                }
+
+                //Since all the resources we care about are freed at this point.
+                GC.SuppressFinalize(this);
+            }
+        }
+
+        #endregion IDisposable Methods
+
+        #region Other Methods
+
+        /// <summary>
+        /// Closes the package and all the underlying parts and relationships.
+        /// Calls the Dispose Method, since they have the same semantics
+        /// </summary>
+        public void Close()
+        {
+            ((IDisposable)this).Dispose();
+        }
+
+        /// <summary>
+        /// Flushes the contents of the parts and the relationships to the package.
+        /// This method will call the FlushCore method which will do the actual flushing of contents.
+        /// </summary>
+        /// <exception cref="ObjectDisposedException">If this Package object has been disposed</exception>
+        /// <exception cref="IOException">If the package is readonly, it cannot be modified</exception>
+        public void Flush()
+        {
+            ThrowIfObjectDisposed();
+            ThrowIfReadOnly();
+
+            // Flush core properties.
+            // Write core properties.
+            // This call will write out the xml for the core properties to the stream
+            // These properties will get flushed to disk as a part of the DoFlush operation
+            if (_packageProperties != null)
+                _packageProperties.Flush();
+
+            // Write package relationships XML to the relationship part stream.
+            // These will get flushed to disk as a part of the DoFlush operation
+            FlushRelationships(); // Flush into .rels part.
+
+            //Write out the Relationship XML for the parts
+            //These streams will get flushed in the DoFlush operation.
+            DoOperationOnEachPart(DoWriteRelationshipsXml);
+
+            // Flush all the parts that are currently open.
+            // This will flush part relationships.
+            DoOperationOnEachPart(DoFlush);
```

Comment:
yeap, I was wrong initially but we can
`DoOperationOnEachPart((thisRef, packagePart) => thisRef.DoFlush(packagePart));`
and pass `this` as parameter inside `DoOperationOnEachPart`.
All these changes related to private/internal API.